src/utils.ts

```diff
@@ -25,4 +25,18 @@ const AsyncFunctionConstructor = (async () => {}).constructor;
 /**
  * an async function check method only consider runtime support async syntax
  */
-export const isAsyncFunction = (fn: Fn) => fn.constructor === AsyncFunctionConstructor;
+export const isAsyncFunction = (fn: Fn) => {
+  if (fn.constructor === AsyncFunctionConstructor) {
+    return true;
+  }
+  try {
+    const call = fn();
```

Comment:
wouldn't that produce side effects?

Reply:
Well, the function is designed to be called a large number of times, right?

Reply:
Do you have one, do you have an example?